notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1205 @@
+      <td>98.75</td>
+    </tr>
+    <tr>
+      <th>0.995</th>
+      <td>4.0</td>
+      <td>111.000</td>
+      <td>131.285</td>
+      <td>112.0</td>
+      <td>214.000</td>
+      <td>192.80</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+For each of these numerical columns, we can look at which rows fall outside of these quantiles and what values they take on. Consider that we want our model to apply to only healthy and mature donkeys.
+
+
+```python
+donkeys[(donkeys['BCS'] < 1.5) | (donkeys['BCS'] > 4)]['BCS']
+```
+
+
+
+
+    291    4.5
+    445    1.0
+    Name: BCS, dtype: float64
+
+
+
+Also looking at the barplot of `BCS`:
+
+
+```python
+plt.hist(donkeys['BCS'], normed=True)
+plt.xlabel('BCS');
+```
+
+
+![png](linear_case_study_files/linear_case_study_17_0.png)
+
+
+Considering that `BCS` is an indication of the health of a donkey, a `BCS` of 1 represents an extremely emaciated donkey and a `BCS` of 4.5 an overweight donkey. Also looking at the barplot, there only appear to be two donkeys with such outlier `BCS` values. Thus, it would probably be a good idea to remove these two donkeys.
+
+---
+
+
+```python
+donkeys[(donkeys['Length'] < 71.145) | (donkeys['Length'] > 111)]['Length']
+```
+
+
+
+
+    8       46
+    22      68
+    26      69
+    216    112
+    Name: Length, dtype: int64
+
+
+
+
+```python
+donkeys[(donkeys['Girth'] < 90) | (donkeys['Girth'] > 131.285)]['Girth']
+```
+
+
+
+
+    8       66
+    239    132
+    283    134
+    523    134
+    Name: Girth, dtype: int64
+
+
+
+
+```python
+donkeys[(donkeys['Height'] < 89) | (donkeys['Height'] > 112)]['Height']
+```
+
+
+
+
+    8       71
+    22      86
+    244    113
+    523    116
+    Name: Height, dtype: int64
+
+
+
+For `Girth`, `Height`, and `Length`, the donkey in row 8 seems to have a much smaller value than the cut-off while the other anomalous donkeys are close to the cut-off and likely do not need to be removed.
+
+---
+
+
+```python
+donkeys[(donkeys['Weight'] < 71.715) | (donkeys['Weight'] > 214)]['Weight']
+```
+
+
+
+
+    8       27
+    26      65
+    50      71
+    291    227
+    523    230
+    Name: Weight, dtype: int64
+
+
+
+The first 2 and last 2 donkeys in the list are far off from the cut-off and most likely should be removed. The middle donkey can be included.
+
+---
+
+Since `WeightAlt` closely corresponds to `Weight`, we do not have to check this column for anomalies. Summarizing what we have learned, here is how we want to filter our donkeys:
+
+- Keep donkeys with `BCS` in the range 1.5 and 4
+- Keep donkeys with `Weight` between 71 and 214  
+
+
+
+```python
+donkeys_c = donkeys[(donkeys['BCS'] >= 1.5) & (donkeys['BCS'] <= 4) &
+                         (donkeys['Weight'] >= 71) & (donkeys['Weight'] <= 214)]
+```
+
+## Train-Test Split
+
+Before we proceed with our data analysis, we divide our data into an 80/20 split, using 80% of our data to train our model and setting aside the other 20% for evaluation of the model.
```

Comment:
hmm i think you're not supposed to look at the testing dataset in any way because it may influence how you create your model :O 

mmm... i don't remember what she said, but her donkeys jupyter notebook just did a 80-20 split. perhaps i can ask her about it tmrw.